### ISSUE TITLE
DeprecationWarning

### ISSUE THREAD

**Issue by dimitri0us** [constverum/ProxyBroker#96]
```
Versions/3.7/lib/python3.7/site-packages/proxybroker/providers.py:128: DeprecationWarning: 'with await lock' is deprecated use 'async with lock' instead
  with (await self._sem_provider),\
```
Working but i got this message on top !

**Comment by erm3nda** [comment]
Fyi warnings don't break usage. Literally means this will stop to work soon
or later.

It may happend that devs do fix code before it breaks. If you don't update
module it will keep working with warning unless you manually supress that
warning.

Regards.

**Comment by constverum** [comment]
Fixed in v0.3.2
